test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -84,31 +84,34 @@ public WorkspaceComparisonData(WorkspaceModel workspace, EngineController contro
                 {
                     NodeTypeMap.Add(n.GUID, n.GetType());
 
-                    var portvalues = new List<object>();
-                    foreach (var p in n.OutPorts)
-                    {
-                        var value = n.GetValue(p.Index, controller);
-                        if (value.IsCollection)
-                        {
-                            portvalues.Add(GetStringRepOfCollection(value));
-                        }
-                        else
-                        {
-                            portvalues.Add(value.StringData);
-                        }
-                    }
-                    
+                    var portvalues = n.OutPorts.Select(p =>
+                        GetDataOfValue(n.GetValue(p.Index, controller))).ToList<object>();
+
                     NodeDataMap.Add(n.GUID, portvalues);
                     InportCountMap.Add(n.GUID, n.InPorts.Count);
                     OutportCountMap.Add(n.GUID, n.OutPorts.Count);
                 }
             }
         }
 
-        private static string GetStringRepOfCollection(ProtoCore.Mirror.MirrorData collection)
+        private static object GetDataOfValue(ProtoCore.Mirror.MirrorData value)
         {
-            var items = string.Join(",", collection.GetElements().Select(x => x.IsCollection ? GetStringRepOfCollection(x) : x.StringData));
-            return "{" + items + "}";
+            if (value.IsCollection)
+            {
+                return value.GetElements().Select(x => GetDataOfValue(x)).ToList<object>();
+            }
+
+            if (!value.IsPointer)
+            {
```

Comment:
I'm not exactly sure.  I was looking for a way to make integer, float, boolean, and string come out as their base types rather than converting them to strings using the StringData function.  Since the Data function handles those the way that I want, but handles pointer differently than I want, I added this check in case a pointer value ended up here.  I don't know that it will actually be hit.